runners/java-fn-execution/src/test/java/org/apache/beam/runners/fnexecution/jobsubmission/JobInvocationTest.java

```diff
@@ -125,6 +128,39 @@ public void testNoCancellationWhenDone() throws Exception {
     assertThat(pipelineResult.cancelLatch.getCount(), is(1L));
   }
 
+  @Test(timeout = 10_000)
+  public void testReturnsMetricsFromJobInvocationAfterSuccess() throws Exception {
+    JobApi.MetricResults expectedMonitoringInfos = JobApi.MetricResults.newBuilder().build();
+    TestPipelineResult result =
+        new TestPipelineResult(PipelineResult.State.DONE, expectedMonitoringInfos);
+
+    jobInvocation.start();
+    runner.setResult(result);
+
+    awaitJobState(jobInvocation, JobApi.JobState.Enum.DONE);
+
+    assertThat(
+        jobInvocation.getMetrics(),
+        allOf(is(notNullValue()), is(sameInstance(result.monitoringInfos()))));
+  }
+
+  @Test(timeout = 10_000)
+  public void testReturnsMetricsFromJobInvocationAfterCancellation() throws Exception {
+    JobApi.MetricResults expectedMonitoringInfos = JobApi.MetricResults.newBuilder().build();
+    TestPipelineResult result =
+        new TestPipelineResult(PipelineResult.State.FAILED, expectedMonitoringInfos);
+
+    // asdasdasd
```

Comment:
Oups! Lol sorry - when I was running tests, this was one of my trials of invalidating gradle cache so that the tests actually run (no idea why running tests is cached in beam). I'm glad that you caught that.